### ISSUE TITLE
Accessibility - file access button group is not keyboard navigable

### ISSUE THREAD

**Issue by kaitlinnewson** [IQSS/dataverse#8247]
**What steps does it take to reproduce the issue?**

On a dataset page, navigate the page using only the keyboard (via the tab key). Note that the file button group for file access can't be accessed via keyboard. **This is a critical accessibility issue** as this button would not be accessible to many users of assistive tech, including screen readers or any keyboard-only user.

**When does this issue occur?**

When navigating the dataset page with a keyboard.

**Which page(s) does it occurs on?**

The dataset page, in the file list.

**What happens?**

The Access File button group can't be accessed via keyboard tabbing.

**To whom does it occur (all users, curators, superusers)?**

All users (anyone using the keyboard).

**What did you expect to happen?**

The button should be keyboard accessible. Part of the issue may be that it is not a proper `<button>` element.

**Which version of Dataverse are you using?**

Tested in 5.8 (Harvard & Harvard demo).

**Screenshot**

The button group that can't be accessed via keyboard:

<img width="968" alt="The file access button" src="https://user-images.githubusercontent.com/6903515/142643740-8616307d-ea02-4672-bc02-0bff010867d1.png">

**Related resources**

https://webaim.org/techniques/keyboard/

https://www.buttoncheatsheet.com/

**Comment by qqmyers** [comment]
@kaitlinnewson / all - FYI: I've been doing a lot of accessibility work for QDR recently that I hope to turn into PRs. The issues that have been documented are in the [QDR fork issues](https://github.com/QualitativeDataRepository/dataverse/issues) and [codebase](https://github.com/QualitativeDataRepository/dataverse). We don't yet have the code deployed to a server outside the VPN/firewalled ones but hope to soon.
To avoid duplicate work, one could scan the list of issues now and see if I've commented on it that there's a fix. (If so we can prioritize getting it to the community code or I can help someone find the fix in the QDR code to transfer adapt). Once we get the code deployed for production, it will be easier - try the same test at QDR and if the issue is resolved, we'll know there's code to move over.
FWIW: There are some issues that appear to be related to the way PrimeFaces generates HTML (e.g. for tabs, some lists, etc.). My guess is that those will require us or PrimeFaces to make PF code changes.

**Comment by qqmyers** [comment]
Also FWIW: most of the issues we've looked at are found by Deque's Axe.

**Comment by qqmyers** [comment]
With a quick check on the QDR dev server - we haven't fixed this specific issue, so if you have a fix, we'll try to grab it.

**Comment by djbrooke** [comment]
Thanks @kaitlinnewson for the report and @qqmyers / QDR for the attention on accessibility issues. @kaitlinnewson if you don't have a fix we will work with @qqmyers to get one in place.

**Comment by kaitlinnewson** [comment]
No, I don't have a fix - just doing some pre-upgrade testing on 5.8. 

Using an icon button instead may work as it appears to render a proper button element, but I'm not very familiar with Primefaces. (e.g. https://www.primefaces.org/showcase/ui/button/button.xhtml?jfwid=bd19d)

**Comment by qqmyers** [comment]
FWIW: There were a couple issues: not only was the ability to tab disabled, once you could focus on the submenu item, hitting a key did nothing (i.e. the submenu didn't fly out). I've fixed those and also addressed a related accessibility issue: fly-out menus should not close immediately per https://www.w3.org/WAI/tutorials/menus/flyout/#mouse-users .

Also - this issue applies to all fly-outs - the dataset edit/permissions, dataset/publish/change curation status, and the auxiliary file download in the file access menu (and the compute button if enabled).

PR to follow.

**Comment by kcondon** [comment]
There may be a temporary workaround for downloading -tab to the filename, hit return, tab to access menu, select from that list.
